lib/fs/util.go

```diff
@@ -228,3 +229,22 @@ func isVolumeNameOnly(parts []string) bool {
 	isUNCVolumeName := len(parts) == 4 && strings.HasSuffix(parts[3], ":")
 	return isNormalVolumeName || isUNCVolumeName
 }
+
+// ReadLines is a utility to read a file at path in filesystem and returns
+// returns its contents as a string-slice of lines.
+func ReadLines(filesystem Filesystem, path string) ([]string, error) {
+	fd, err := filesystem.Open(path)
+	if err != nil {
+		return nil, err
+	}
+	defer fd.Close()
+	scanner := bufio.NewScanner(fd)
+	var lines []string
+	for scanner.Scan() {
+		lines = append(lines, scanner.Text())
+	}
```

Comment:
Should we somehow limit the number of lines?  It's kind of a DoS attack vector to put a huge file as .stignore and then trying to add the folder will possibly jam Syncthing be eating all RAM?

Reply:
There's no remote action involved, just the local user running a cli command. And what's the difference of opening the file in an editor or here, memory wise?

Reply:
Guess the only difference is that the file is is not being consciously opened, but a side effect of adding a Syncthing folder. Of course it's unlikely that someone else has put a huge file with that name somewhere for others to trip over.

Reply:
This is only used when loading ignores from a file on disk to config.xml with `syncthing cli`. I also removed this function and put the code directly into `.../cli/operations.go`, I think I originally created this util function in earlier versions of this PR when had been used in multiple places.

Reply:
Ah sorry my confusion.